examples/activities/delegator_activity.rb

```diff
@@ -0,0 +1,35 @@
+# This sample illustrates using a dynamic Activity to delegate to another set of non-activity
+# classes.  This is an advanced use case, used, for example, for integrating with an existing framework
+# that doesn't know about temporal.
+# See Concerns::Executable#dynamic for more info.
+
+# An example of another non-Activity class hierarchy.
+class MyExecutor
+  def do_it(args)
+    raise NotImplementedError.new
+  end
+end
+
+class Plus < MyExecutor
+  def do_it(args)
+    args[:a] + args[:b]
+  end
+end
+
+class Times < MyExecutor
+  def do_it(args)
+    args[:a] * args[:b]
+  end
+end
+
+# Calls into our other class hierarchy.
+class DelegatorActivity < Temporal::Activity
+  dynamic
+
+  def execute(input)
+    executor = Object.const_get(activity.name).new
```

Comment:
I think it's a bit confusing that `activity.name` here is different to the activity itself. I would prefer the name of the "sub-activity" to be passed in as the first argument to this activity. I almost view this activity as Ruby's [`method_missing`](https://ruby-doc.org/core-2.6.3/BasicObject.html#method-i-method_missing)

Reply:
added some comments and undid some unnecessary changes to the `Activity::Context` constructor (the name's already in the metadata)